src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionListener.cs

```diff
@@ -26,6 +26,11 @@ internal class QuicConnectionListener : IMultiplexedConnectionListener, IAsyncDi
 
         public QuicConnectionListener(QuicTransportOptions options, IQuicTrace log, EndPoint endpoint)
         {
+            if (options.Alpn == null)
+            {
+                throw new InvalidOperationException("QuicTransportOptions.Alpn must be configured with a value.");
```

Comment:
What should happen if no ALPN is configured?

Reply:
I think this is fine, I'm not sure what we can put as a default.

Reply:
@jkotalik 

I think this is good enough for now. The whole experience needs to be improved:

https://github.com/dotnet/aspnetcore/blob/9cfb13398c3296f51114a4b9eb01d69b93b5195c/src/Servers/Kestrel/samples/Http3SampleApp/Program.cs#L28-L33

It also looks like the IdleTimeout defaults to zero which probably isn't ideal.